### PR TITLE
Fix docker kill

### PR DESCRIPTION
**Pull request by jonasfj**

### CODE REVIEW

File: engines/docker/engine_test.go [hunk 1]
```diff
@@ -4,8 +4,15 @@ import (
...
-	"github.com/taskcluster/taskcluster-worker/engines/enginetest"
 	"time"
+
+	"github.com/taskcluster/taskcluster-worker/engines/enginetest"
+)
+
+// Image and tag used in test cases below
+const (
+	dockerImageRepository = "alpine"
+	dockerImageTag        = "3.6"
```
Comment (jonasfj) [inline engines/docker/engine_test.go]: The image hash disappeared from the repository without warning... So I guess maybe we have to use tags...
Or well, at-least we might have to support both.

File: engines/docker/sandbox.go [hunk 2]
```diff
@@ -98,59 +98,107 @@ func (s *sandbox) wait() {
...
+
+	// Send SIGTERM and give the container 30s to exit
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel() // always free the context
+	err := s.client.KillContainer(docker.KillContainerOptions{
+		ID:      s.containerID,
+		Signal:  docker.SIGTERM,
+		Context: ctx,
+	})
+	// Report errors if ContainerNotRunning or ctx was timed out
```
Comment (walac) [inline engines/docker/sandbox.go]: This comment conflicts with the function description.

File: engines/docker/sandbox.go [hunk 3]
```diff
@@ -98,59 +98,107 @@ func (s *sandbox) wait() {
...
+	if _, ok := err.(*docker.ContainerNotRunning); err != nil && !ok && ctx.Err() == nil {
+		s.monitor.ReportError(err, "KillContainer with SIGTERM failed")
+		// signal up the stack that something went wrong, this is not a successful kill
+		hasErr = true
```
Comment (walac) [inline engines/docker/sandbox.go]: I think it is better if we add a name to the return value and set it here.

File: engines/docker/sandbox.go [hunk 4]
```diff
@@ -98,59 +98,107 @@ func (s *sandbox) wait() {
...
+	// Free image handle
+	s.handle.Release()
+
+	// Remove temporary storage
+	if err = s.tempStorage.Remove(); err != nil {
+		s.monitor.ReportError(err, "failed to remove temporary storage in disposal of sandbox")
+	}
+
+	// If ErrNonFatalInternalError if there was an error of any kind
+	if hasErr {
```
Comment (walac) [inline engines/docker/sandbox.go]: `hasErr` is never set to `true`

**Pull request closed without merging**